packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -387,7 +393,7 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
     final Widget prefix = Padding(
       child: IconTheme(
         data: iconThemeData,
-        child: const Icon(CupertinoIcons.search),
+        child: widget.prefixIcon,
```

Comment:
I thing you need to update the docus of the `enabled` property to make it clear the the `prefixIcon` will not respond to touch events.

See https://github.com/flutter/flutter/pull/79966#discussion_r610371491 for context.
 Cc @LongCatIsLooong

Reply:
I also updated the documentation of `CupertinoSearchTextField`.